src/js/actions/deviceActions.js

```diff
@@ -806,3 +811,16 @@ export const applyDeviceConfig = (deviceId, configDeploymentConfiguration, isDef
       }
       return Promise.all(tasks);
     });
+
+export const setDeviceTags = (deviceId, tags) => dispatch =>
+  // to prevent tag set failures, retrieve the device & use the freshest etag we can get
+  Promise.resolve(dispatch(getDeviceById(deviceId))).then(device => {
+    const headers = device.etag ? { 'If-Match': device.etag } : {};
```

Comment:
The `etag` is not in the device object, but in the `etag` HTTP header.
Because of this, the UI is not sending the `If-Match` header on PUT.

Reply:
[the `getDeviceById` call](https://github.com/mendersoftware/gui/pull/1814/files/0727e9298b913f573392b2a9b8ffbeb0b8ebe878#diff-74f29c7fa70ecbddf5cd2b41184645057374578e6b26e9a4ff4ff49d3cabb157R402) sets the etag on the device and [returns it](https://github.com/mendersoftware/gui/pull/1814/files/0727e9298b913f573392b2a9b8ffbeb0b8ebe878#diff-74f29c7fa70ecbddf5cd2b41184645057374578e6b26e9a4ff4ff49d3cabb157R405) to be able to rely on an always up-to-date etag (as it might soon become stale if the UI relied on the storage)... - so if the etag is present in the response from inventory it _should_ be sent during PUT

Reply:
Oh, sorry, my fault. When testing locally, it never sent the `If-Match` header, though.